themes/src/main/resources/theme/base/login/template.ftl

```diff
@@ -151,4 +151,20 @@
   </div>
 </body>
 </html>
+
+<script>
+    function preventDoubleClick(){
+        const submitButtons = document.querySelectorAll('input[type="submit"]');
+        for (var i = 0; i < submitButtons.length; i++) {
+            const submitButton = submitButtons[i];
+            submitButton.disabled = true;
+        }
+        return true;
+    }
```

Comment:
The [`SubmitEvent`](https://developer.mozilla.org/en-US/docs/Web/API/SubmitEvent) has a special property called [`submitter`](https://developer.mozilla.org/en-US/docs/Web/API/SubmitEvent/submitter) that can be used to determine the element that submitted the form. We can use this element instead of re-querying the DOM.

```suggestion
    function preventDoubleClick({ submitter }) {
        if (submitter !== null) {
            submitter.disabled = true;
        }
    }
```

Although this code only disables the submitted form I think that is enough for most cases where the user might accidentally 'double click'.

Reply:
As i see Submitter it's quite new feature and the most browsers start support it recently.
Chrome since v80 (2020), Safari sine 15.4 (2022)  https://caniuse.com/?search=SubmitEvent.submitter
Do you have agreements regarding browsers and their versions that you support?

Reply:
Yes, that is within our browser support target so it should not be an issue. If you like it would be possible to add a graceful degradation by changing the check to `submitter !== null && submitter !== undefined`.

Reply:
Done